src/components/Map/Map.tsx

```diff
@@ -483,316 +515,289 @@ const Map = ({
-              'source-layer': 'raccordements',
-              ...raccordementsLayerStyle,
-            },
-          ]
-        );
-        setMapState('loaded');
-      }
-    });
-
-    map.current.on('load', () => {
-      map.current.loadImage(
-        '/icons/rect.png',
-        (error: any, image: Record<string, unknown>) => {
-          if (error) {
-            throw error;
-          }
-
-          setMapState('loaded');
-          map.current.addImage('energy-picto', image, { sdf: true });
-
-          if (!noPopup) {
-            clickEvents.map(({ name, key }) => {
-              map.current.on('click', name, (e: any) => {
-                onMapClick(e, key);
-              });
-
-              map.current.on('touchend', name, (e: any) => {
-                onMapClick(e, key);
-              });
-
-              map.current.on('mouseenter', name, function () {
-                map.current.getCanvas().style.cursor = 'pointer';
-              });
-
-              map.current.on('mouseleave', name, function () {
-                map.current.getCanvas().style.cursor = '';
-              });
-            });
-          }
-
-          addHover(map, 'heatNetwork', 'outline');
-          addHover(map, 'heatFuturNetwork', 'futurOutline');
-          addHover(map, 'coldNetwork', 'coldOutline');
-
-          // ----------------
-          // --- Controls ---
-          // ----------------
-          const navControl = new maplibregl.NavigationControl({
-            showCompass: true,
-            showZoom: true,
-            visualizePitch: true,
-          });
-          map.current.addControl(navControl, 'top-left');
-
-          const attributionControl = new maplibregl.AttributionControl({
-            compact: false,
-          });
-          map.current.addControl(attributionControl, 'bottom-right');
+        ]
+      );
 
-          const scaleControl = new maplibregl.ScaleControl({
-            maxWidth: 100,
-            unit: 'metric',
-          });
-          map.current.addControl(scaleControl, 'bottom-left');
-        }
+      // -----------------
+      // --- Raccordements ---
+      // -----------------
+      addSource(
+        e.target,
+        'raccordements',
+        {
+          type: 'vector',
+          tiles: [`${origin}/api/map/raccordements/{z}/{x}/{y}`],
+          maxzoom: maxZoom,
+          minzoom: minZoomData,
+        },
+        [
+          {
+            id: 'raccordements',
+            source: 'raccordements',
+            'source-layer': 'raccordements',
+            ...raccordementsLayerStyle,
+          },
+        ]
       );
-    });
-  });
+      setMapState('loaded');
+    }
+  };
 
   useEffect(() => {
     const { id } = router.query;
-    if (!id) {
+    if (!id || mapRef.current == null) {
       return;
     }
 
     heatNetworkService.bulkEligibilityValues(id as string).then((response) => {
-      if (response.result) {
+      if (response.result && mapRef.current) {
```

Comment:
```suggestion
      if (response.result) {
```
tu checkes deja le current plus haut